kvdb-rocksdb/src/lib.rs

```diff
@@ -666,12 +679,17 @@ impl Database {
 	}
 
 	/// Add a column family.
-	pub fn add_column(&self) -> io::Result<()> {
+	pub fn add_column(&mut self) -> io::Result<()> {
 		match *self.db.write() {
 			Some(DBAndColumns { ref mut db, ref mut cfs }) => {
-				let col = cfs.len() as u32;
+				let col = cfs.len();
 				let name = format!("col{}", col);
-				cfs.push(db.create_cf(&name, &col_config(&self.config, &self.block_opts)?).map_err(other_io_err)?);
+				if self.config.memory_budget.len() < col + 1 {
+					self.config.memory_budget.resize(col + 1, None);
+				}
+				let memory_budget_per_col = self.config.memory_budget_per_col(col);
+				let col_config = col_config(&self.config, &self.block_opts, memory_budget_per_col)?;
```

Comment:
Couldn't `col_config()` be a method of `DatabaseConfig` and have access to it's own config and be able to do all the setup there? Same for line [335](https://github.com/paritytech/parity-common/pull/256/files#diff-4382bfb7e75959b2b8884268540abfa3R335)?